lib/std/c.zig

```diff
@@ -5,29 +5,25 @@ const page_size = std.mem.page_size;
 const iovec = std.os.iovec;
 const iovec_const = std.os.iovec_const;
 
-/// The return type is `type` to force comptime function call execution.
-/// TODO: https://github.com/ziglang/zig/issues/425
 /// If not linking libc, returns struct{pub const ok = false;}
 /// If linking musl libc, returns struct{pub const ok = true;}
 /// If linking gnu libc (glibc), the `ok` value will be true if the target
 /// version is greater than or equal to `glibc_version`.
 /// If linking a libc other than these, returns `false`.
-pub fn versionCheck(comptime glibc_version: std.SemanticVersion) type {
-    return struct {
-        pub const ok = blk: {
-            if (!builtin.link_libc) break :blk false;
-            if (builtin.abi.isMusl()) break :blk true;
-            if (builtin.target.isGnuLibC()) {
-                const ver = builtin.os.version_range.linux.glibc;
-                const order = ver.order(glibc_version);
-                break :blk switch (order) {
-                    .gt, .eq => true,
-                    .lt => false,
-                };
-            } else {
-                break :blk false;
-            }
-        };
+pub inline fn versionCheck(comptime glibc_version: std.SemanticVersion) bool {
+    return comptime blk: {
+        if (!builtin.link_libc) break :blk false;
```

Comment:
Might be more straight-forward to remove the `blk:` tags/break and just `return false` or `return true` in each branch?

Reply:
it makes a function call which would make that branch runtime known, the comptime block is used to force the entire expression to be comptime